ethereum/contracts/TablelandRigPilots.sol

```diff
@@ -279,6 +279,50 @@ contract TablelandRigPilots is
         emit Training(tokenId);
     }
 
+    /**
+     * @dev See {ITablelandRigPilots-pilotRig}.
+     */
+    function pilotRig(address sender, uint256 tokenId) external onlyParent {
+        // Validate the Rig can be piloted with a trainer pilot. Note that
+        // `_canPilot` allows for in-flight piloting once trained *and* in
+        // `TRAINING` status, but this only allows for piloting a with trainer
+        // if `PARKED`.
+        if (_pilotStatus(tokenId) != GarageStatus.PARKED)
+            revert InvalidPilotStatus();
+
+        // Assign a trainer pilot to the Rig in `_pilots`. The pilot contract is
+        // `0` and pilot ID `2`—this is needed to differentiate from a
+        // `TRAINING` Rig's pilot, which has a pilot ID of `1`.
+        _setPilotData(uint16(tokenId), 0, 2);
+
+        // Set the start time for the new pilot session
+        _setStartTime(uint16(tokenId), uint64(block.number));
+
+        // Insert the trainer pilot into the Tableland pilot sessions table
+        TablelandDeployments.get().runSQL(
+            address(this),
+            _pilotSessionsTableId,
+            SQLHelpers.toInsert(
+                _PILOT_SESSIONS_PREFIX,
+                _pilotSessionsTableId,
+                "rig_id,owner,pilot_contract,pilot_id,start_time",
+                string.concat(
+                    StringsUpgradeable.toString(uint16(tokenId)),
+                    ",",
+                    SQLHelpers.quote(StringsUpgradeable.toHexString(sender)),
+                    ",",
+                    SQLHelpers.quote(Strings.toHexString(address(0))),
+                    ",",
+                    StringsUpgradeable.toString(uint32(2)),
+                    ",",
+                    StringsUpgradeable.toString(uint64(block.number))
+                )
```

Comment:
`trainRig` doesn't write a `pilot_contract` or `pilot_id` value to the pilot sessions table so the data will look pretty different between a training session and a non-training session without an NFT pilot.

If we want to think of training and flying without an NFT pilot as using two separate pilots I think this makes sense, I think we should probably set the contract value to null and not 0x0 to be consistent with the training session but that is a minor detail. And we would have to update the garage to know that it shouldn't try to fetch NFT metadata for the collection `{ contract: 0x0, id: 2 }` and that we should hardcode that as "Default Pilot" instead of "Trainer" or something like that

But if we don't want to think of it as two separate pilots I think this is a suboptimal solution, working with the data would be a bit annoying since you can no longer do a simple
```sql
SELECT pilot_contract, pilot_id, count(*) FROM pilot_sessions WHERE rig_id = ?id GROUP BY pilot_contract, pilot_id
```
to get the number of flight sessions per pilot for instance. In that case I think we should either add a column to the pilot_sessions table to indicate what type of session it is (training/not training) and use the same contract and id (which would be annoying since we don't support migrations), or add another table called something like pilot_session_type where we store `(pilot_session_id, type)` tuples

Does that make sense?

Reply:
@datadanne would the garage require changes if we simply didn't write `pilot_contract` and `pilot_id` to the pilots table when a rig that was `PARKED` goes flying with the trainer? ie, in `function pilotRig(address sender, uint256 tokenId) external onlyParent` do:

```solidity
TablelandDeployments.get().runSQL(
    address(this),
    _pilotSessionsTableId,
    SQLHelpers.toInsert(
        _PILOT_SESSIONS_PREFIX,
        _pilotSessionsTableId,
        "rig_id,owner,start_time",
        string.concat(
            StringsUpgradeable.toString(uint16(tokenId)),
            ",",
            SQLHelpers.quote(StringsUpgradeable.toHexString(sender)),
            ",",
            StringsUpgradeable.toString(uint64(block.number))
        )
    )
);
```

Then, from the garage perspective, when you pilot with the trainer by clicking "pilot", you can treat it like it like normal training. One hiccup I can think of is how to know if the rig can be piloted from training. This is how the contract determines if this is possible:

```
// Cannot switch real pilots mid-flight, but can go from trainer -> real pilot
// There are two ways to pilot:
//   1. In a `PARKED` status
//   2. In a `TRAINING` status for long enough (30 days; 172800 blocks)
```

So, you'd have to determine if the rig has previously completed a training session, which I think you already do by querying for at least one completed session. You'd have to grey out the pilot button when the trainer is in use in this case. That might be confusing, but not too bad.

Reply:
(I'm fine with `NULL` pilot contract/ID values for any type of trainer pilot session—just lmk what to adjust here)

I didn't want to complicate things for this fix but had this broader feeling: a trainer pilot _is_ different than this "stock" default pilot. We _could_ make the stock pilot a Rig-bound token...nothing fancy with the token itself (e.g., same table setup as Rigs but basically all empty with a stock pilot token ID, one default attribute, and same rendering as the trainer but a different color). That bound stock pilot could be upgraded to something else in the future. Plus, that solves for the `0x0` and default `2` design since here, there would be a real pilot contract and unique pilot ID.

It's out of scope and wanted to throw the idea out there...but short term, not feasible.

Reply:
sg. let's see what @datadanne says about my suggestion above.

I like the stock pilot idea... worth exploring the idea in a notion doc at some point?

Reply:
Threw some ideas down here—not an official proposal but wanted to put it down in writing https://www.notion.so/textile/Rigs-stock-default-pilot-b88de504639448bfb98079a25805a6a8?pvs=4

Reply:
> @datadanne would the garage require changes if we simply didn't write pilot_contract and pilot_id to the pilots table when a rig that was PARKED goes flying with the trainer? ie, in function pilotRig(address sender, uint256 tokenId) external onlyParent do:

Nope, all the pilot-related code in the garage can stay the same if we leave the `pilot_contract` and `pilot_id` values empty when an already trained rig is piloted with the trainer pilot again. 

We still need to update the garage to allow a user to take a rig in-flight that is trained using the pilot, right now we only support using an ERC721 as the pilot if the rig is trained.

> ie, with @dtbuchholz 's change, the contract would still know about this distinction: TRAINING vs. PILOTED, which is used to determine if the rigs gets reset to UNTRAINED during a force park. But, i'm not sure the garage needs to know about that distinction at all.

Yeah the garage doesn't need to know this at all as long as rigs are reset to UNTRAINED when they get force parked in the TRAINING state. 

We will still need to look into how many rigs are in this inconsistent state right now though, if there are many that got force parked during training we still have to handle that in the garage even though the contract won't allow new rigs to get into that state

Reply:
sg. @dtbuchholz, sounds like this PR will be good if you make the change here: https://github.com/tablelandnetwork/rigs/pull/720#discussion_r1154679035

Reply:
@sanderpick I figure I should upgrade the mumbai contracts in this PR, too. With the changes to `upgradeRigs.ts` in https://github.com/tablelandnetwork/rigs/commit/024d569e8e8df1b63a6296d7ed65cef5e91beaee, if/how does that impact how I should upgrade rigs on mumbai? Should I ignore that `TMP` section when running the script?